pygfx/utils/show.py

```diff
@@ -34,33 +44,47 @@ def show(object: WorldObject, up=None):
 
         background = Background(None, BackgroundMaterial((0, 1, 0, 1), (0, 1, 1, 1)))
         scene.add(background)
+        scene.add(AmbientLight(0.2))
 
-    camera = PerspectiveCamera(70, 16 / 9)
-    look_at = camera.show_object(object)
-    scene.add(camera)
+    if not camera:
+        camera = PerspectiveCamera(70, 16 / 9)
+        camera.add(DirectionalLight(0.8))
+        scene.add(camera)
 
-    # Add lights if there are none
-    light_count = 0
+    if canvas is None:
+        from wgpu.gui.auto import WgpuCanvas, run
 
-    def check_light(ob):
-        nonlocal light_count
-        if isinstance(ob, Light):
-            light_count += 1
+        canvas = WgpuCanvas()
+        event_loop = run
+    elif event_loop is None:
+        # TODO: find the matching run function
+        raise ValueError(
+            "When providing a canvas you also need to provide the event loop."
+        )
 
-    scene.traverse(check_light, False)
-    if not light_count:
-        camera.add(DirectionalLight(0.8))
-        scene.add(AmbientLight(0.2))
+    if renderer is None:
+        renderer = WgpuRenderer(canvas)
```

Comment:
A custom canvas should be disallowed if a custom renderer is given. We don't really need the canvas here, we can do the `request_draw()` call below on the renderer instead.

Reply:
Sure.